### ISSUE TITLE
Cannot open developer tools after upgrading to beta 1.4.0

### ISSUE THREAD

**Issue by pleonex** [atom/atom#10003]
After upgrading to the latest beta version I am not longer able to open the develper tools. It just does nothing. I am seeing also random crashes but not sure if it's related.

**Reproduction Steps:**

1. Open Atom in any mode, e.g.: `atom --foreground --safe`
2. Try to toggle the developer tools. Same behavior from `View -> Developer -> Toggle Developer Tools` or using the shortcut `Ctrl+Shift+I`

Same behavior if starting Atom in dev mode.

**Errors logged**
* When trying to toggle the developer tools.
```
[24901:1210/161108:INFO:CONSOLE(22)] "Uncaught SyntaxError: Unexpected identifier", source: /usr/local/share/atom/resources/atom.asar/common/api/lib/callbacks-registry.js (22)
[24901:1210/161108:INFO:CONSOLE(52)] "Uncaught SyntaxError: Unexpected identifier", source: /usr/local/share/atom/resources/app.asar/node_modules/notifications/lib/main.js (52)
```
* Full log opening in dev mode
```
benito@fedora22-benito:~$ atom --foreground --dev
[25305:1210/161638:ERROR:browser_main_loop.cc(189)] Running without the SUID sandbox! See https://code.google.com/p/chromium/wiki/LinuxSUIDSandboxDevelopment for more information on developing with the sandbox on.
[25305:1210/161639:WARNING:accelerator_util.cc(118)] Invalid accelerator token: command
[25305:1210/161639:WARNING:accelerator_util.cc(118)] Invalid accelerator token: command
[25305:1210/161639:WARNING:accelerator_util.cc(118)] Invalid accelerator token: command
[25305:1210/161639:WARNING:accelerator_util.cc(118)] Invalid accelerator token: command
[25337:1210/161639:ERROR:renderer_main.cc(200)] Running without renderer sandbox
auto-updater is not implemented on this platform
Error: auto-updater is not implemented on this platform
    at Object.<anonymous> (/usr/local/share/atom/resources/atom.asar/browser/api/lib/auto-updater.js:11:13)
    at Object.<anonymous> (/usr/local/share/atom/resources/atom.asar/browser/api/lib/auto-updater.js:14:4)
    at Module._compile (module.js:434:26)
    at Object.defineProperty.value [as .js] (/usr/local/share/atom/resources/app.asar/src/compile-cache.js:204:21)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at AutoUpdateManager.module.exports.AutoUpdateManager.setupAutoUpdater (/usr/local/share/atom/resources/app.asar/src/browser/auto-update-manager.js:51:23)
    at /usr/local/share/atom/resources/app.asar/src/browser/auto-update-manager.js:42:24
    at doNTCallback0 (node.js:419:9)
    at process._tickCallback (node.js:348:13)
    at NavigationController.loadUrl (/usr/local/share/atom/resources/atom.asar/browser/api/lib/navigation-controller.js:59:24)
    at EventEmitter.webContents.(anonymous function) [as loadUrl] (/usr/local/share/atom/resources/atom.asar/browser/api/lib/web-contents.js:81:27)
    at BrowserWindow.loadUrl (/usr/local/share/atom/resources/atom.asar/browser/api/lib/browser-window.js:121:37)
    at AtomWindow.module.exports.AtomWindow.setLoadSettings (/usr/local/share/atom/resources/app.asar/src/browser/atom-window.js:124:33)
    at new AtomWindow (/usr/local/share/atom/resources/app.asar/src/browser/atom-window.js:110:12)
    at AtomApplication.module.exports.AtomApplication.openPaths (/usr/local/share/atom/resources/app.asar/src/browser/atom-application.js:686:24)
    at AtomApplication.module.exports.AtomApplication.openWithOptions (/usr/local/share/atom/resources/app.asar/src/browser/atom-application.js:132:21)
    at AtomApplication.module.exports.AtomApplication.loadState (/usr/local/share/atom/resources/app.asar/src/browser/atom-application.js:770:16)
    at new AtomApplication (/usr/local/share/atom/resources/app.asar/src/browser/atom-application.js:114:14)
    at createAtomApplication (/usr/local/share/atom/resources/app.asar/src/browser/atom-application.js:60:16)
    at Function.module.exports.AtomApplication.open (/usr/local/share/atom/resources/app.asar/src/browser/atom-application.js:63:9)
    at EventEmitter.<anonymous> (/usr/local/share/atom/resources/app.asar/src/browser/main.js:55:23)
    at emitOne (events.js:82:20)
    at EventEmitter.emit (events.js:169:7)
App load time: 629ms
[25305:1210/161643:INFO:CONSOLE(0)] "::shadow pseudo-element is deprecated. See https://www.chromestatus.com/features/6750456638341120 for more details.", source:  (0)
[25305:1210/161643:INFO:CONSOLE(713)] "Failed to require the main module of 'term3' because it requires an incompatible native module.
Run `apm rebuild` in the package directory to resolve.", source: /usr/local/share/atom/resources/app.asar/src/package.js (713)
[25305:1210/161644:INFO:CONSOLE(51)] "Window load time: 4432ms", source: file:///usr/local/share/atom/resources/app.asar/static/index.js (51)
[25305:1210/161644:INFO:CONSOLE(0)] "/deep/ combinator is deprecated. See https://www.chromestatus.com/features/6750456638341120 for more details.", source:  (0)
[25305:1210/161644:INFO:CONSOLE(166)] "Uncaught (in promise) SyntaxError: Unexpected identifier", source: /usr/local/share/atom/resources/app.asar/src/compile-cache.js (166)
[25305:1210/161644:INFO:CONSOLE(166)] "Uncaught (in promise) SyntaxError: Unexpected identifier", source: /usr/local/share/atom/resources/app.asar/src/compile-cache.js (166)
[25305:1210/161644:INFO:CONSOLE(448)] "Could not resolve 'highlight-selected' to a package path", source: /usr/local/share/atom/resources/app.asar/src/package-manager.js (448)
[25305:1210/161645:WARNING:accelerator_util.cc(118)] Invalid accelerator token: command
[25305:1210/161645:WARNING:accelerator_util.cc(118)] Invalid accelerator token: command
[25305:1210/161645:INFO:CONSOLE(368)] "Uncaught (in promise) [object Array]", source: node.js (368)
[25305:1210/161653:INFO:CONSOLE(658)] "Uncaught SyntaxError: Unexpected identifier", source: /usr/local/share/atom/resources/app.asar/src/package.js (658)
```

**Atom version:** 1.4.0-beta0-ed7283f
**OS and version:** Fedora 22 with KDE running in VirtualBox 5.

**Installed packages:**

```
atom-material-ui
autocomplete-clang
autocomplete-emojis
autocomplete-python
autocomplete-xml
blame
docblockr
file-icons
fold-comments
git-control
git-diff-details
git-history
git-log
git-plus
git-projects
hyperclick
idiomatic-dark-syntax
linter
linter-alex
linter-clang
linter-gcc
linter-pep8
linter-xmllint
merge-conflicts
minimap
minimap-autohide
minimap-bookmarks
minimap-find-and-replace
minimap-git-diff
minimap-highlight-selected
minimap-linter
minimap-selection
tag
term3
tool-bar
tool-bar-almighty
travis-ci-status
xkcd-comics
```

**Additional information:**

* Problem can be reproduced in safe mode: Yes
* Problem started happening recently, didn't happen in an older version of Atom: Yes, after upgrading to beta 1.4.0.
* Problem can be reliably reproduced, doesn't happen randomly: Yes, it happens always.
* Problem happens with all files and projects, not only some files or projects: Yes

**Comment by 50Wliu** [comment]
Thanks for the detailed bug report!  I haven't experienced this on Windows so I'm going to label it as a Linux bug.

**Comment by pleonex** [comment]
Same error trying to open [Git Control](https://atom.io/packages/git-control)
```
[25474:1210/170414:INFO:CONSOLE(1039)] "Uncaught SyntaxError: Unexpected identifier", source: /usr/local/share/atom/resources/app.asar/src/package.js (1039)
[25474:1210/170414:INFO:CONSOLE(52)] "Uncaught SyntaxError: Unexpected identifier", source: /usr/local/share/atom/resources/app.asar/node_modules/notifications/lib/main.js (52)
```

**Comment by pleonex** [comment]
Fixed after removing the *.atom/blob-store* and *.atom/compile-cache* folders.
I have a backup copy of the original folders, let me know if you want some info to investigate it.

**Comment by clistenes** [comment]
Thank you @pleonex, for me it worked the removal of folders.
Using Mac OS 10.11.2

**Comment by nathansobo** [comment]
@pleonex If you have a chance could you narrow down whether it's the blob store or the compile cache or both that are causing the issue?

**Comment by pleonex** [comment]
@nathansobo @as-cii you were right, it's fixed after just removing *.atom/blob-store/*.

**Comment by andya9** [comment]
Thank’you @pleonex , your fix worked here too

**Comment by nathansobo** [comment]
@andya9 What's your platform?

**Comment by jdiaz5513** [comment]
Saw this too, OSX 10.10.5, was likely instigated by (accidentally) running atom stable and atom beta simultaneously. Removing the blob-store folder fixed it.

**Comment by as-cii** [comment]
Interesting. So, I investigated this issue today morning and I think we've just found an edge case in the v8 compile cache that triggers this behavior.

I have documented it in https://github.com/atom/cached-run-in-this-context/commit/ff1671c570c9153eb1e2a3055c039dfadc93323d: when compiling two different functions that are sufficiently similar, v8 cannot distinguish anymore between them and is not able to reject the cache.

This means we need to address this issue in core by hashing the content of files and use that as a key for later cache retrievals. Working on that now.
